### PR TITLE
Hugo marcel/oauth

### PR DESCRIPTION
**Pull request by dkhgh**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,6 +21,8 @@ gem "sass-rails", "~> 5.0"
 gem "simple_form"
 gem "title"
 gem "uglifier"
+gem 'omniauth'
+gem 'omniauth-yammer'
```
Comment (fatboypunk) [inline Gemfile]: `"` :dog:

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class User < ActiveRecord::Base
+  validates :name, presence: true
+  validates :email, presence: true
+  validates :uid, presence: true
+  validates :provider, presence: true
+
+  def self.create_with_omniauth(auth)
+    create! do |user|
+      pp auth
```
Comment (fatboypunk) [inline app/models/user.rb]: Piew Piew

File: config/routes.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 Rails.application.routes.draw do
+	root to: "static_pages#home"
```
Comment (fatboypunk) [inline config/routes.rb]: tab :dog:

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (fatboypunk) [inline spec/controllers/users_controller_spec.rb]: we can remove the controller specs

File: spec/features/logout_spec.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+require 'spec_helper'
+require 'rails_helper'
+
+feature "log out" do
+  scenario "user logs out" do
+    visit register_path
+    set_omniauth(opts = {})
```
Comment (fatboypunk) [inline spec/features/logout_spec.rb]: I think we can create a user, set the session user_id to the users id and than log out without registering?

File: spec/features/static_pages_spec.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+require 'rails_helper'
+
+feature "static home page" do
+  
+  scenario "anonymous user visits the home page" do
+    visit root_url
+    expect(page).to have_content("Welcome to Chief Awesome!")
+  end
+end
```
Comment (fatboypunk) [inline spec/features/static_pages_spec.rb]: Time to update your sublime config ^^

Comment (dkhgh) [inline spec/features/static_pages_spec.rb]: auto newline?

**Comment by dkhgh** [comment]
@fatboypunk think I fixed everything

File: spec/support/session_helper.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+def sign_in_user
+  visit register_path
+  set_omniauth(opts = {})
+  click_link "Register with Yammer"
+end
```
Comment (fatboypunk) [inline spec/support/session_helper.rb:5]: :dog2:

**Pull request merged**